### PR TITLE
modify loading tokens and erc20 txs

### PR DESCRIPTION
**Pull request by jingweicb**
problem: 
fix: adding whiteListtokens to configuratuion

minor modifications:
changed rosetta-geth-dependence

adding an new type:invalid

### CODE REVIEW

**Comment by jingweicb** [comment]
@mdehoog please have a look of this pr

File: config.go [hunk 1]
```diff
@@ -179,6 +179,10 @@ func LoadConfiguration() (*configuration.Configuration, error) {
 	}
 
 	tokenListJson := os.Getenv(TokenListEnv)
+	// if users do not provide tokens, read from the internal files
+	if tokenListJson == "" {
+		tokenListJson = "tokenList.json"
+	}
```
Comment (mdehoog) [inline config.go]: i don't think we should default to anything... let's just use the envvar

File: tokenList.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[
+    {
+        "chainId": 3222583904,
```
Comment (mdehoog) [inline tokenList.json]: this is not applicable to other chains... i don't think we should include this file in this repo (we can always show an example in the README)

**Comment by jingweicb** [comment]
@mdehoog thanks for your comments, modified

File: main.go [hunk 3]
```diff
@@ -14,17 +14,11 @@ func main() {
...
-			continue
-		}
-		ots = append(ots, ot)
-	}
-	ots = append(ots, MintOpType)
+	t.OperationTypes = append(types.OperationTypes, types.InvalidOpType)
+	// Optimism does not have miner or uncle reward type ops
 	// TODO(inphi): add BurnOpType
-	t.OperationTypes = ots
+	t.OperationTypes = t.OperationTypes[2:]
```
Comment (mdehoog) [inline main.go]: This seems like a style regression... it's always assuming that miner and uncle will be the first two options in the slice. This is why i initially added the nice explicit filtering loop above.

File: go.mod [hunk 4]
```diff
@@ -3,8 +3,8 @@ module github.com/mdehoog/op-rosetta
 go 1.18
 
 require (
-	github.com/coinbase/rosetta-geth-sdk v0.0.4
-	github.com/coinbase/rosetta-sdk-go v0.7.11
+	github.com/coinbase/rosetta-geth-sdk v0.0.5-0.20230124171712-bed30e6be8c7
```
Comment (mdehoog) [inline go.mod:6]: should we do a release rather than relying on the sha?

File: main.go (hunk 3 above)
Comment (jingweicb) [reply main.go]: gotcha

File: go.mod (hunk 4 above)
Comment (jingweicb) [reply go.mod:6]: last time, we modified our sdk according to an asset very frequently, if we keep releasing new versions, perhaps there will be a lot of future releases, but I am ok to both choice.

Comment (jingweicb) [reply go.mod:6]: @mdehoog not this sdk, we modified an internal sdk according to an blockchain very frequently

**Review by mdehoog — APPROVED** [review]

**Pull request merged**